### PR TITLE
ci: trigger packages builds only manually or with a comment

### PR DESCRIPTION
**Pull request by kuisathaverat**
## What does this PR do?

it puts a filter in place to avoid allocate and agent if the build is not triggered by a comment or a user.

## Why is it important?

it avoids waste resources.

## Checklist

- ~~[ ] My code follows the style guidelines of this project~~
- ~~[ ] I have commented my code, particularly in hard-to-understand areas~~
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~
- ~~[ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~~

### CODE REVIEW

File: .ci/packaging.groovy [hunk 1]
```diff
@@ -3,7 +3,7 @@
 @Library('apm@current') _
 
 pipeline {
-  agent { label 'ubuntu && immutable' }
+  agent none
```
Comment (v1v) [inline .ci/packaging.groovy:6]: will it work the `gitCheckout` ?

Comment (kuisathaverat) [reply .ci/packaging.groovy:6]: I have tested and it works

Comment (kuisathaverat) [reply .ci/packaging.groovy:6]: However, I have launched it on the CI to test it also in the CI https://beats-ci.elastic.co/job/Beats/job/packaging/job/PR-19559/2/console, I'll merge it when it finished

File: .ci/packaging.groovy [hunk 2]
```diff
@@ -12,6 +12,7 @@ pipeline {
     DOCKERELASTIC_SECRET = 'secret/observability-team/ci/docker-registry/prod'
     DOCKER_REGISTRY = 'docker.elastic.co'
     SNAPSHOT = "true"
+    PIPELINE_LOG_LEVEL = "DEBUG"
```
Comment (kuisathaverat) [inline .ci/packaging.groovy:15]:
```suggestion
    PIPELINE_LOG_LEVEL = "INFO"
```

**Pull request merged**